python/rmm/tests/test_rmm.py

```diff
@@ -719,3 +719,25 @@ def test_dev_buf_circle_ref_dealloc():
     # deallocate `dbuf1` (which needs the MR alive), a segfault occurs.
 
     gc.collect()
+
+
+def test_custom_mr(capsys):
+    base_mr = rmm.mr.CudaMemoryResource()
+
+    def allocate_func(size):
+        print(f"Allocating {size} bytes")
+        return base_mr.allocate(size)
+
+    def deallocate_func(ptr, size):
+        print(f"Deallocating {size} bytes")
+        return base_mr.deallocate(ptr, size)
```

Comment:
Depending on whether we decide to include exposing `allocate` and `deallocate` as part of this PR, the alternative would probably be to use CUDA Python's `cuda.cuda.cuMemAlloc`.

Reply:
I think exposing those is probably a good idea anyway because typically you want to "extend" an existing MR with `CallbackMemoryResource`.